src/directives.ts

```diff
@@ -0,0 +1,124 @@
+import { PluginOptions } from './main';
+
+
+interface DestroyHTMLElement extends HTMLElement {
+    $destroy: () => void;
+}
+
+interface DestroyHTMLInputElement extends HTMLInputElement {
+  $destroy: () => void;
+}
+
+/**
+ * Triggers action when scroll gets close to the bottom.
+ *
+ * Example usage:
+ * <div v-infinite-scroll="action()">Scroll me</div>
+ */
+const infiniteScroll: DirectiveOptions = {
+  bind(el: DestroyHTMLElement, binding: DirectiveBinding): void {
+    const tolerance = parseInt(binding.arg || '100', 10);
+    const action = binding.value;
+
+    el.addEventListener('scroll', (): void => {
+      const height = (el.firstChild as DestroyHTMLElement).clientHeight;
+      if (el.scrollTop >= height - el.clientHeight - tolerance) {
+        action();
+      }
+    });
+  },
+};
+
+/**
+ * Scroll input into view when focused.
+ *
+ * Example usage:
+ * <input type="text" value="foo" v-scroll-on-focus />
+ */
+const scrollOnFocus: DirectiveOptions = {
+  bind(el: DestroyHTMLInputElement): void {
+    const target = el.querySelector('input,textarea');
+    if (!target) {
+      return;
+    }
+
+    const scrollTo = (target: HTMLElement, scrollable: HTMLElement, scrollableContent: HTMLElement, marginTop: number = 0): void => {
+      while (target.offsetParent && target.offsetParent !== scrollable) {
+        marginTop -= target.offsetTop;
+        target = target.offsetParent as HTMLElement;
+      }
+      scrollableContent.scrollTop = target.offsetTop - marginTop;
+    };
+
+    const handler = (): void => {
+      const scrollable = getClosest(el, '.scrollable');
+      const scrollableContent = getClosest(el, '.scrollable-content');
+      if (scrollable && scrollableContent) {
+        const height = scrollableContent.offsetHeight;
+        setTimeout((): void => {
+          // if scrollableContent height is reduced in half second
+          // since an input got focus we assume soft keyboard is showing.
+          if (height > scrollableContent.offsetHeight) {
+            scrollTo(target as HTMLElement, scrollable, scrollableContent, 20);
+          }
+        }, 500);
+      }
+    };
+
+    target.addEventListener('focus', handler);
+    el.$destroy = (): void => {
+      target.removeEventListener('focus', handler);
+    };
+  },
+  unbind(el: DestroyHTMLElement): void {
+    el.$destroy();
+  },
+};
+
+export type trackClickHandlerFunction = (eventName: string | undefined, eventProps: {[key: string]: any}) => void; // eslint-disable-line @propelinc/no-explicit-any
+var trackClickHandler: trackClickHandlerFunction;
+/**
+ * Track an event to amplitude on click.
+ *
+ * Example usage:
+ * <div v-track-click event-name="foo" :event-props="{ bar: 'tzar' }">Click me</div>
+ */
+const trackClick: DirectiveOptions =  {
+  bind(el: DestroyHTMLElement, binding: DirectiveBinding, vnode: VNode): void {
+    const attrs = vnode.data && vnode.data.attrs ? vnode.data.attrs: {};
+
+    function wrappedHandler(): void {
```

Comment:
This is the type of function where an arrow function is recommended. It reduces complexity by not introducing a new this.

Reply:
good point